### PR TITLE
Leaky Faucet: Rework faucet for latest contracts, plus… More.

### PR DESCRIPTION
**Pull request by Shadowfiend**
The commits lay it out, but high-level:

- Better output (text/plain response type + more messages).
- Don't wait for the transaction to mine.
- Send an etherscan link.
- Update for 0.13.0 by adding a staking policy to the `approveAndCall` invocation.

I might have rewritten most of the thing by accident while doing this <_<

### CODE REVIEW

File: infrastructure/gcp/keep-test/google-functions/keep-faucet/issue-grant.js [hunk 1]
```diff
@@ -39,72 +38,117 @@ const web3 = new Web3(keepContractOwnerProvider, null, web3Options)
...
+              You can find us on Discord at: https://discord.gg/jqxBU4m .\n
+            `)
+          })
+          .on('error', (error) => {
+            console.error(
+              `Error with account grant transaction: [${error}]; URL was [${request.url}].`
+            )
+            if (! response.headersSent) {
+              response.status(500).send(`
+                  Token grant failed, try again.\n
```
Comment (sthompson22) [inline infrastructure/gcp/keep-test/google-functions/keep-faucet/issue-grant.js:138]: Is there a reason the `500` responses are tabbed more than 2 spaces?

File: infrastructure/gcp/keep-test/google-functions/keep-faucet/issue-grant.js [hunk 2]
```diff
@@ -39,72 +38,117 @@ const web3 = new Web3(keepContractOwnerProvider, null, web3Options)
...
-        )} to: ${granteeAccount}`,
-      )
-      response.send(`
-        Created token grant with ${web3.utils.toBN(
-          grantAmount,
-        )} KEEP for account: ${granteeAccount}
-        You can manage your token grants at: https://dashboard.test.keep.network
-        You can find us on Discord at: https://discord.gg/jqxBU4m\n`)
+        console.log("Test submission...")
+        // Try calling; if this throws, we'll have a proper error message thanks
```
Comment (sthompson22) [inline infrastructure/gcp/keep-test/google-functions/keep-faucet/issue-grant.js:109]: This is neat.

File: infrastructure/gcp/keep-test/google-functions/keep-faucet/issue-grant.js [hunk 3]
```diff
@@ -39,72 +38,117 @@ const web3 = new Web3(keepContractOwnerProvider, null, web3Options)
...
+
+        console.log("Submitting transaction...")
+        // If the call didn't revert, try submitting the transaction proper.
+        keepToken.methods
+          .approveAndCall(tokenGrantAddress, grantAmount, grantData)
+          .send({ from: keepContractOwnerAddress })
+          .on('transactionHash', (hash) => {
+            console.log(
+              `Submitted grant for [${grantAmount}] to [${granteeAccount}].`,
+              transaction,
```
Comment (sthompson22) [inline infrastructure/gcp/keep-test/google-functions/keep-faucet/issue-grant.js:123]: I'm not sure where `transaction` comes from here?  Should this be `hash`?

Comment (sthompson22) [reply infrastructure/gcp/keep-test/google-functions/keep-faucet/issue-grant.js:123]: Went ahead and tested this, the function barfs on `transaction` not being defined, but `hash` does seem to work. 

Example error.

```
Serving function...
Function: issueGrant
URL: http://localhost:8080/
Fetching existing balance for account [0xcCb2acbC6db2432d577b9EdBD42c1B125ef9925e]...
Existing balance for account [0xcCb2acbC6db2432d577b9EdBD42c1B125ef9925e] is [0].
Submitting grant for [300000000000000000000000] to [0xcCb2acbC6db2432d577b9EdBD42c1B125ef9925e]...
Test submission...
Submitting transaction...
ReferenceError: transaction is not defined
    at Promise.keepToken.methods.approveAndCall.send.on (/Users/sthompson22/Projects/keep/keep-core/infrastructure/gcp/keep-test/google-functions/keep-faucet/issue-grant.js:123:15)
    at Promise.emit (/Users/sthompson22/Projects/keep/keep-core/infrastructure/gcp/keep-test/google-functions/keep-faucet/node_modules/eventemitter3/index.js:181:35)
    at sendTxCallback (/Users/sthompson22/Projects/keep/keep-core/infrastructure/gcp/keep-test/google-functions/keep-faucet/node_modules/web3/node_modules/web3-core-method/src/index.js:625:36)
    at /Users/sthompson22/Projects/keep/keep-core/infrastructure/gcp/keep-test/google-functions/keep-faucet/node_modules/web3/node_modules/web3-core-requestmanager/src/index.js:147:9
    at /Users/sthompson22/Projects/keep/keep-core/infrastructure/gcp/keep-test/google-functions/keep-faucet/node_modules/web3-provider-engine/index.js:159:9
    at /Users/sthompson22/Projects/keep/keep-core/infrastructure/gcp/keep-test/google-functions/keep-faucet/node_modules/async/internal/once.js:12:16
    at replenish (/Users/sthompson22/Projects/keep/keep-core/infrastructure/gcp/keep-test/google-functions/keep-faucet/node_modules/async/internal/eachOfLimit.js:61:25)
    at /Users/sthompson22/Projects/keep/keep-core/infrastructure/gcp/keep-test/google-functions/keep-faucet/node_modules/async/internal/eachOfLimit.js:71:9
    at eachLimit (/Users/sthompson22/Projects/keep/keep-core/infrastructure/gcp/keep-test/google-functions/keep-faucet/node_modules/async/eachLimit.js:43:36)
    at /Users/sthompson22/Projects/keep/keep-core/infrastructure/gcp/keep-test/google-functions/keep-faucet/node_modules/async/internal/doLimit.js:9:16
    at end (/Users/sthompson22/Projects/keep/keep-core/infrastructure/gcp/keep-test/google-functions/keep-faucet/node_modules/web3-provider-engine/index.js:134:5)
    at /Users/sthompson22/Projects/keep/keep-core/infrastructure/gcp/keep-test/google-functions/keep-faucet/node_modules/async/internal/once.js:12:16
    at next (/Users/sthompson22/Projects/keep/keep-core/infrastructure/gcp/keep-test/google-functions/keep-faucet/node_modules/async/waterfall.js:21:29)
    at /Users/sthompson22/Projects/keep/keep-core/infrastructure/gcp/keep-test/google-functions/keep-faucet/node_modules/async/internal/onlyOnce.js:12:16
    at /Users/sthompson22/Projects/keep/keep-core/infrastructure/gcp/keep-test/google-functions/keep-faucet/node_modules/async/internal/once.js:12:16
    at next (/Users/sthompson22/Projects/keep/keep-core/infrastructure/gcp/keep-test/google-functions/keep-faucet/node_modules/async/waterfall.js:21:29)
```

Interesting that even after ^ is thrown the account does get a grant anyway.

File: infrastructure/gcp/keep-test/google-functions/keep-faucet/issue-grant.js [hunk 4]
```diff
@@ -39,72 +38,117 @@ const web3 = new Web3(keepContractOwnerProvider, null, web3Options)
...
+            )
+            response.send(`
+              Created token grant with ${grantAmount} KEEP for account: ${granteeAccount}\n
+              You can follow the transaction at https://ropsten.etherscan.io/tx/${transaction.hash}\n
```
Comment (sthompson22) [inline infrastructure/gcp/keep-test/google-functions/keep-faucet/issue-grant.js:127]: Same deal as comment above this.

File: infrastructure/gcp/keep-test/google-functions/keep-faucet/issue-grant.js (hunk 1 above)
Comment (Shadowfiend) [reply infrastructure/gcp/keep-test/google-functions/keep-faucet/issue-grant.js:138]: That is a good question. Lemme make sure prettier hit this.

File: infrastructure/gcp/keep-test/google-functions/keep-faucet/issue-grant.js (hunk 2 above)
Comment (Shadowfiend) [reply infrastructure/gcp/keep-test/google-functions/keep-faucet/issue-grant.js:109]: I both agree and desperately hate it lol.

File: infrastructure/gcp/keep-test/google-functions/keep-faucet/issue-grant.js (hunk 3 above)
Comment (Shadowfiend) [reply infrastructure/gcp/keep-test/google-functions/keep-faucet/issue-grant.js:123]: Howwww did I do that, I tested this 🤨

File: infrastructure/gcp/keep-test/google-functions/keep-faucet/issue-grant.js (hunk 1 above)
Comment (Shadowfiend) [reply infrastructure/gcp/keep-test/google-functions/keep-faucet/issue-grant.js:138]: Ok, prettiered.

File: infrastructure/gcp/keep-test/google-functions/keep-faucet/issue-grant.js (hunk 3 above)
Comment (Shadowfiend) [reply infrastructure/gcp/keep-test/google-functions/keep-faucet/issue-grant.js]: Fix't.

File: infrastructure/gcp/keep-test/google-functions/keep-faucet/issue-grant.js (hunk 4 above)
Comment (Shadowfiend) [reply infrastructure/gcp/keep-test/google-functions/keep-faucet/issue-grant.js]: Samesies.

**Pull request merged**